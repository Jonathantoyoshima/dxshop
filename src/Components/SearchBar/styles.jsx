import styled from 'styled-components';

export const Root = styled.div`
    align-items: center;
    box-sizing: border-box;
    display: flex;
    margin-bottom: 10px;
    padding: 30px;
    width: 100%;
    @media only screen and (max-width: 600px) {
        flex-direction: column;
    }
`;

export const Text = styled.div`
    white-space: nowrap;
    @media only screen and (min-width: 600px) {
        margin-right: 30px;
    }    

`;
