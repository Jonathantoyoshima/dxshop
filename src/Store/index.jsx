import React from 'react';

const categories = [
  { id: 0, name: 'Bebidas' },
  { id: 1, name: 'Doces' },
  { id: 2, name: 'Salgados' },
];

const data = [
  { id: 0, idCategory: 0, name: 'Coca-cola lata 350ml', price: 3.5 },
  { id: 1, idCategory: 0, name: 'Fanta lata 350ml', price: 3.5 },
  { id: 2, idCategory: 1, name: 'Bolinho Ana maria', price: 1.5 },
  { id: 3, idCategory: 2, name: 'Ruffes Pequeno 350g', price: 5.0 },
];

const cart = [];

const info = { data: data, categories: categories, cart: cart, chooseCategory: -1 };

export const StoreContext = React.createContext(info);

export const Store = ({ children }) => {
  const [context, setContext] = React.useState(info);

  return <StoreContext.Provider value={[context, setContext]}>{children}</StoreContext.Provider>;
};
