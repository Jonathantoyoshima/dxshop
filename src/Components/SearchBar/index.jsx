import React from 'react';
import { Select } from 'Styles';
import { Root } from './styles';
import { StoreContext } from 'Store';

const SearchBar = () => {
  const [store, setStore] = React.useContext(StoreContext);
  const handleFilter = (e) => {
    setStore({ ...store, chooseCategory: e.currentTarget.value });
  };

  return (
    <Root>
      <Select onChange={handleFilter}>
        <option value={-1}>Selecione aqui uma categoria</option>
        {store.categories.map((category, index) => (
          <option key={index} value={category.id}>
            {category.name}
          </option>
        ))}
      </Select>
    </Root>
  );
};

export default SearchBar;
