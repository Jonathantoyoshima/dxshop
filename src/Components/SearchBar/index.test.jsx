import React from 'react';
import { shallow } from 'enzyme';
import SearchBar from './index';
import { Store } from 'Store';

test('Render Build SearchBar Snapshot', () => {
  const component = shallow(
    <Store>
      <SearchBar />
    </Store>
  );
  expect(component).toMatchSnapshot();
});
