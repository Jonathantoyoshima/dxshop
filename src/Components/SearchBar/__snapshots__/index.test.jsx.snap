// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Build SearchBar Snapshot 1`] = `ShallowWrapper {}`;
